### PR TITLE
291 replace asyncwith

### PR DESCRIPTION
**Pull request by mgordel**

### CODE REVIEW

File: examples/package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
         "commander": "^9.1.0",
         "dayjs": "~1.9.1",
         "ts-node": "^10.7.0",
-        "yajsapi": "^0.6.0"
+        "yajsapi": "file:../dist"
```
Comment (zakaprov) [inline examples/package.json:29]: Should this be updated before merging?

**Review by zakaprov — APPROVED** [review]
:ok:

File: examples/package.json (hunk 1 above)
Comment (mgordel) [reply examples/package.json]: Yes, my mistake

**Comment by shadeofblue** [comment]
one (maybe important) remark that I have is that if we're anyway removing the `asyncWith` and requiring the Executor instance to be created before we're able to use it, maybe we could also provide a start/stop interface where we don't have to limit what we do with that intance to the body of the function passed to `run` ... 

but maybe that can be done separately...

**Review by shadeofblue — APPROVED** [review]
besides the comment above, it looks good

**Pull request merged**